src/runtime/virtcontainers/fc_metrics.go

```diff
@@ -0,0 +1,761 @@
+// Copyright (c) 2020 Ant Financial
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package virtcontainers
+
+import (
+	"encoding/json"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const namespaceFirecracker = "kata_firecracker"
+
+func (fc *firecracker) updateMetrics(line string) {
+	var fm FirecrackerMetrics
+	if err := json.Unmarshal([]byte(line), &fm); err != nil {
+		fc.Logger().WithError(err).WithField("data", line).Error("failed to unmarshal fc metrics")
+		return
+	}
+	updateFirecrackerMetrics(&fm)
+}
+
+//
+// below is codes generated from
```

Comment:
Could you put this auto-generated code in a different file maybe with a warning comment at the top like https://github.com/kata-containers/kata-containers/blob/2.0-dev/src/runtime/cli/config-generated.go.in:

```golang
// WARNING: This file is auto-generated - DO NOT EDIT!
```

Reply:
Yes, it's better. I can put other codes into fc.go, and leave this as auto-generated.